### PR TITLE
Articles read time

### PR DESCRIPTION
**Pull request by milosyuki**
Articles read and watch time

### CODE REVIEW

**Review by robwalsh21 — APPROVED** [review]

File: workspaces/netlify-cms/src/App.tsx [hunk 1]
```diff
@@ -1,8 +1,50 @@
...
+      <input
+        type="hidden"
+        value={value}
+        onChange={(e) => setBodyText(e.target.value)}
+        name={field.get('name')}
+      />
+      <p>Reading time: {value} minutes</p>
+    </div>
+  );
+};
```
Comment (thecotne) [inline workspaces/netlify-cms/src/App.tsx]: this is left over code right?

File: workspaces/netlify-cms-widgets/src/timeToConsume.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  };
+}
+
+export const NetlifyCmsWidgetTimeToConsume = {
+  Widget,
+  controlComponent: Control,
+  previewComponent: Preview,
+};
+
+export default NetlifyCmsWidgetTimeToConsume;
```
Comment (thecotne) [inline workspaces/netlify-cms-widgets/src/timeToConsume.tsx]: this is left over code right?

File: workspaces/netlify-cms-widgets/src/youtube.tsx [hunk 3]
```diff
@@ -86,8 +84,8 @@ function Control(props: CmsWidgetControlProps & any) {
       >
         {fetching
           ? `Fetching: ${fetching}`
-          : title
-          ? `Video: ${title}`
+          : title && formattedDuration
+          ? `Video: ${title} | Duration: ${formattedDuration}`
```
Comment (thecotne) [inline workspaces/netlify-cms-widgets/src/youtube.tsx]: `formattedDuration` is never defined in this file

File: src/app/[locale]/posts/[category]/[slug]/page.tsx [hunk 4]
```diff
@@ -175,18 +179,58 @@ function pageToTableOfContents(page: any): readonly HeadingData[] {
...
 }
 
+function formatDuration(duration: string): string {
+  const match = duration.match(/PT(\d+H)?(\d+M)?(\d+S)?/);
+  if (!match) {
+    return '';
+  }
+  const hours = parseInt(match[1]) || 0;
+  const minutes = parseInt(match[2]) || 0;
+  const totalMinutes = hours * 60 + minutes; // pretvaranje u minute
```
Comment (thecotne) [inline src/app/[locale]/posts/[category]/[slug]/page.tsx]: what's the language of the comments?

File: src/app/[locale]/posts/[category]/[slug]/page.tsx [hunk 5]
```diff
@@ -175,18 +179,58 @@ function pageToTableOfContents(page: any): readonly HeadingData[] {
...
     const category = categories.find((c) => c.id === post.category)!;
 
     let videoId = post.post_type === "video" ? post.video?.id : undefined;
-
+    const fullText = concatenateBodies(post.blocks);
+    let timeToRead;
+    if (post.post_type === "video") {
+      timeToRead = `${formatDuration(post.video?.data?.contentDetails?.duration || '')} watch`;
+    } else {
+      timeToRead = `${calculateReadingTime(fullText)} read`;
```
Comment (thecotne) [inline src/app/[locale]/posts/[category]/[slug]/page.tsx]: `timeToRead` should be calculated in `cms-scripts` (in fileToPost function)

File: src/app/[locale]/posts/(components)/PostsPage.tsx [hunk 6]
```diff
@@ -311,7 +367,13 @@ function CustomHits({ categories }: Pick<Props, "categories">) {
       >
         {hits.map((hit, i) => {
           // todo: add a featured image once we have image templates in place
-
+          const fullText = concatenateBodies(hit.blocks);
+          let timeToRead;
+          if (hit.post_type === "video") {
+            timeToRead = `${formatDuration(hit.video?.data?.contentDetails?.duration || '')} watch`;
+          } else {
+            timeToRead = `${calculateReadingTime(fullText)} read`;
```
Comment (thecotne) [inline src/app/[locale]/posts/(components)/PostsPage.tsx]: too much calculations on each iteration move to `cms-scripts`

**Review by thecotne — CHANGES_REQUESTED** [review]

File: workspaces/netlify-cms/src/App.tsx (hunk 1 above)
Comment (milosyuki) [reply workspaces/netlify-cms/src/App.tsx]: You are right. Thanks, Tsotne, removed.

File: workspaces/netlify-cms-widgets/src/timeToConsume.tsx (hunk 2 above)
Comment (milosyuki) [reply workspaces/netlify-cms-widgets/src/timeToConsume.tsx]: You are right. Thanks, Tsotne, removed.

File: workspaces/netlify-cms-widgets/src/youtube.tsx (hunk 3 above)
Comment (milosyuki) [reply workspaces/netlify-cms-widgets/src/youtube.tsx]: You are right. Thanks, Tsotne, removed.

File: src/app/[locale]/posts/[category]/[slug]/page.tsx (hunk 4 above)
Comment (milosyuki) [reply src/app/[locale]/posts/[category]/[slug]/page.tsx]: Thanks, Tsotne, removed.

File: src/app/[locale]/posts/[category]/[slug]/page.tsx (hunk 5 above)
Comment (milosyuki) [reply src/app/[locale]/posts/[category]/[slug]/page.tsx]: Thanks for the comment, Tsotne, I moved it to fileToPost function.

File: src/app/[locale]/posts/(components)/PostsPage.tsx (hunk 6 above)
Comment (milosyuki) [reply src/app/[locale]/posts/(components)/PostsPage.tsx]: Thanks for the comment, Tsotne, I moved it to fileToPost function.

**Review by thecotne — APPROVED** [review]

**Pull request merged**